plugins/modules/roles.ps1

```diff
@@ -0,0 +1,119 @@
+#!powershell
+# -*- coding: utf-8 -*-
+
+# (c) 2022, John McCall (@lowlydba)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#AnsibleRequires -PowerShell ansible_collections.lowlydba.sqlserver.plugins.module_utils._SqlServerUtils
+#Requires -Modules @{ ModuleName="dbatools"; ModuleVersion="1.1.112" }
+
+$ErrorActionPreference = "Stop"
+
+$spec = @{
+    supports_check_mode = $true
+    options = @{
+        database = @{type = 'str'; required = $true }
+        username = @{type = 'str'; required = $true }
+        roles = @{type = 'list'; elements = 'str'; required = $false }
+        state = @{type = 'str'; required = $false; default = 'present'; choices = @('present', 'absent') }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec, @(Get-LowlyDbaSqlServerAuthSpec))
+$sqlInstance, $sqlCredential = Get-SqlCredential -Module $module
+$username = $module.Params.username
+$database = $module.Params.database
+$roles = $module.Params.roles
+$state = $module.Params.state
+$checkMode = $module.CheckMode
+
+$module.Result.changed = $false
+
+$getRoleSplat = @{
+    SqlInstance = $sqlInstance
+    SqlCredential = $sqlCredential
+    Database = $database
+    EnableException = $true
+}
+$module.Result.roles = $roles
+$existingRoleObjects = Get-DbaDbRoleMember @getRoleSplat | Where-Object { $_.UserName -eq $username }
+$existingRoles = @()
+# build an array of roles for the selected user
+foreach ($roleObject in $existingRoleObjects) {
+    $existingRoles += $roleObject.role
+}
+# Always return a list of existing roles if any exist
+if ($null -ne $existingRoleObjects) {
+    $module.Result.existingRoles = $existingRoles
+}
+else {
+    $module.Result.noRoles = "'$username' doesn't have any existing roles assigned on '$database'"
+}
+
+if ($state -eq "absent") {
+    # loop through all roles to remove and see if any are assigned to the user
+    $removeRoles = @()
+    foreach ($roleObject in $existingRoleObjects) {
+        if ($roles.Contains($roleObject.role)) {
+            $removeRoles += $roleObject.role
+        }
+    }
```

Comment:
I think this can be removed, the DBATools function should gracefully handle removing roles whether they exist or not for the user.

Reply:
Yes. However, if we don't check first there's no way to inform the user which roles are being added or removed.

Reply:
True, but I think that might be best served by the `_info` module. 

I mostly avoided using lists as inputs for this reason - it adds complexity and error handling gets more complicated for partial successes, etc. In my main use of this collection I use loops to run a module for a series of variables which while a bit slower helps to isolate issues more cleanly. It also helps to keep this as close as possible to simply passing through DBATools' results - if they were to start adding the roles modified in their result set, it would get confusing since we're now passing a custom result with a similar/duplicate list. 

Thinking through this some more, it might make the most sense to do a feature request to DBATools to output the roles changed as an object (pretty sure they'd be open to this - feel free to tag me if you do).

Reply:
I was a little surprised to see that those DBATools functions didn't return anything at all.  But I'll go ahead and simplify these modules.  I'll see about doing that feature request as well.